### PR TITLE
Import enhancement

### PR DESCRIPTION
**Pull request by User-green**
This is a system, witch will install missing libs if user agree to do so. 
Some libs not supported due to tham been used in the system. 
Require some updates to make it work with python not been in the PATH or pip been not installed. I need some help with this. 
Also i need to know, if this usefull at all.
@8go, please check this out.

### CODE REVIEW

**Comment by 8go** [comment]
Hi @User-green 

Thank you for your suggested contribution, and all the effort you made. That is very kind of you. 

It is always nice to see people helping out and contributing ideas or code. 

This very specific suggestion has a few drawbacks:
As example look at this line: 
`           "asyncio";[r"https://github.com/python/cpython/tree/3.9/Lib/asyncio/", "none"],`

What are the drawbacks?
- This idea hard-codes URLs of repos into the source code. This is generally a bad idea as repos might move and hence URLs might change.
- This idea version numbers, or repo trees (e.g. `3.9`)  into the source code. This is generally a bad idea as branches and repo trees will change with time, breaking the code.
- This is *NOT* the standard way of doing things or installing packages in Python. Other people wanting to use this app will be surprised and probably not like it because it does not follow a Python "standard", i.e. typical way of doing things.
- Look at `os.popen("python3 -m pip install --no-input -q "+str(module))`. This line is a bit worrisome. A new process is started installing packages in your operating system. This line is dangerous, it could be misused to install a virus or malicious code. Any user would have to look a lot more careful at this code now with this line included to make sure there is no misuse of this specific line. Not a good idea for reasons of security.
- There is a very powerful tool specific for handling installations and requirements, that is `pip`. So we should use it, instead of recreating part of it inside our source code. A singe line like `pip install -r requirements.txt` usually does *all* the installation. And the user does this *CONSCIOUSLY* being aware that he is installing new packages to the OS.

In short, I feel that your code makes
- the app more dependent on URLs and version numbers
- does things in a non-standard way
- is less secure, or at least opens possibilities for misuse to install unwanted packages

Sorry for the bad news, please @User-green  don't take this as personal criticism, it is just a comment that I give with good intentions.

**Comment by User-green** [comment]
@8go, I see that you are totaly right! Even more - I see, how I can fix listed issues and make code simpler!

I will update this request\close in a few days. 
Only one question. Where can i find references for "standart"?
 
 
 P.S. Links was not used in actual installation. I thought, that it will be nice to include references to the libs been used.
 If links got broken - nothing bad will happen. Thought - in updated wersion of this i will write tham as separete file, and only will link root of the reop.

**Comment by 8go** [comment]
> Only one question. Where can i find references for "standart"?

Well, there is not 1 specific document. It is more the form that Python usage has evolved over time. 
If you read different Python books you will find the same pattern over and over:
- source code has `import somepackage`
- github repo has: `requirements.txt` file containing `somepackage`
- user runs installation as: `pip install -r requirements.txt`

You will find this specific pattern in all books. Have a look at all of your favorite Python repos on Github, they will all look like this! Hence it is a "standard" as this is was programmers do, what users expect. It is simple, easy to understand.

It is simpler and a clean separation of "code" and "installation". The code should not do the installation.

Additional files (e.g. holding dependency lists, etc) will make the app *LESS* simple. And most likely it will make it so, that the same or related info is now in 2 places; making it difficult to maintain because now 2 places have to be changed ...

Providing URLs (while well intended for reference) do *NOT* help as they need to be maintained or will be out of date soon. 
Everyone knows to look at https://pypi.org/ if one wants info (version, location, source code, etc) of a Python package. E,g, https://pypi.org/project/matrix-nio/ Here everything is up-to-date and digitally signed (secure). That is why package info does not go into the source code. It is on https://pypi.org/.


Again, @User-green no criticism intended. Just my viewpoint.

**Comment by User-green** [comment]
Thank you, this was helpfull. 
Expect issues been fixed in range from one three days to three mounth.  
P.S. Constructive criticism is always wellcome. I have to be critisised, if there is my fault. I have to make lessons from mistakes. Ofcours it would be better to use others mistakes as reference, but if I make one - that meant only that I need to know it and not make same in the future.

**Comment by User-green** [comment]
Closing this. I will do it eventualy, but not in near future.
